src/platform_impl/linux/wayland/window/mod.rs

```diff
@@ -338,6 +337,11 @@ impl CoreWindow for Window {
         super::logical_to_physical_rounded(window_state.outer_size(), scale_factor)
     }
 
+    fn safe_area(&self) -> (PhysicalPosition<u32>, PhysicalSize<u32>) {
+        // FIXME: Include CSDs drawn by Winit
```

Comment:
Hmm, should we then update `surface_position` to include the CSDs drawn by Winit?

Reply:
On Wayland, I think it won't change anything since CSDs are in negative part of the window, so the surface is still at (0,0), it's just CSDs are in (-border, -header), so don't have to do anything, since it's in the (0,0).

Reply:
Wow, that's wild! How does the compositor allow you to create and draw surfaces outside the bounds that you are otherwise restricted to?